pkg/host/udev.go

```diff
@@ -0,0 +1,171 @@
+package host
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/consts"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/vars"
+)
+
+type config struct {
+	Interfaces []sriovnetworkv1.Interface `json:"interfaces"`
+}
+
+func (h *hostManager) PrepareNMUdevRule(supportedVfIds []string) error {
+	log.Log.V(2).Info("PrepareNMUdevRule()")
+	filePath := filepath.Join(vars.FilesystemRoot, consts.HostUdevRulesFolder, "10-nm-unmanaged.rules")
+
+	// remove the old unmanaged rules file
+	if _, err := os.Stat(filePath); err == nil {
+		err = os.Remove(filePath)
+		if err != nil {
+			log.Log.Error(err, "failed to remove the network manager global unmanaged rule",
+				"path", filePath)
+		}
+	}
+
+	// create the pf finder script for udev rules
+	stdout, stderr, err := h.utilsHelper.RunCommand("/bin/bash", filepath.Join(vars.FilesystemRoot, consts.UdevDisableNM))
+	if err != nil {
+		log.Log.Error(err, "PrepareNMUdevRule(): failed to prepare nmUdevRule", "stderr", stderr)
+		return err
+	}
+	log.Log.V(2).Info("PrepareNMUdevRule()", "stdout", stdout)
+
+	//save the device list to use for udev rules
+	vars.SupportedVfIds = supportedVfIds
+	return nil
+}
+
+func (h *hostManager) WriteSwitchdevConfFile(newState *sriovnetworkv1.SriovNetworkNodeState, pfsToSkip map[string]bool) (update bool, err error) {
+	cfg := config{}
+	for _, iface := range newState.Spec.Interfaces {
+		for _, ifaceStatus := range newState.Status.Interfaces {
+			if iface.PciAddress != ifaceStatus.PciAddress {
+				continue
+			}
+
+			if skip := pfsToSkip[iface.PciAddress]; !skip {
+				continue
+			}
+
+			i := sriovnetworkv1.Interface{}
+			if iface.NumVfs > 0 {
+				var vfGroups []sriovnetworkv1.VfGroup = nil
+				ifc, err := sriovnetworkv1.FindInterface(newState.Spec.Interfaces, iface.Name)
+				if err != nil {
+					log.Log.Error(err, "WriteSwitchdevConfFile(): fail find interface")
+				} else {
+					vfGroups = ifc.VfGroups
+				}
+				i = sriovnetworkv1.Interface{
+					// Not passing all the contents, since only NumVfs and EswitchMode can be configured by configure-switchdev.sh currently.
+					Name:       iface.Name,
+					PciAddress: iface.PciAddress,
+					NumVfs:     iface.NumVfs,
+					Mtu:        iface.Mtu,
+					VfGroups:   vfGroups,
+				}
+
+				if iface.EswitchMode == sriovnetworkv1.ESwithModeSwitchDev {
+					i.EswitchMode = iface.EswitchMode
+				}
+				cfg.Interfaces = append(cfg.Interfaces, i)
+			}
+		}
+	}
+	_, err = os.Stat(consts.SriovHostSwitchDevConfPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if len(cfg.Interfaces) == 0 {
+				err = nil
+				return
+			}
+
+			// Create the sriov-operator folder on the host if it doesn't exist
+			if _, err := os.Stat("/host" + consts.SriovConfBasePath); os.IsNotExist(err) {
```

Comment:
this is from container right? 
so /host is the host itself.
so then it falls under /etc/sriov..... ?

Can we make "/host" a const somewhere? this will be easier later in cleaning up host imports.

Reply:
I didn't change this function at all I put a TODO on a following PR this one is already huge :)